ledger-core/virtual/execute/execute_test.go

```diff
@@ -488,3 +491,68 @@ func TestSMExecute_VCallResultPassedToSMObject(t *testing.T) {
 
 	mc.Finish()
 }
+
+func TestSendVStateReportWithMissingState_IfConstructorWasInterruptedBeforeRunnerCall(t *testing.T) {
+	t.Log("C5084")
+	var (
+		mc  = minimock.NewController(t)
+		ctx = context.Background()
+
+		class                              = gen.UniqueGlobalRef()
+		caller                             = gen.UniqueGlobalRef()
+		catalog     object.Catalog         = object.NewLocalCatalog()
+		authService authentication.Service = authentication.NewServiceMock(t)
+	)
+
+	slotMachine := slotdebugger.New(ctx, t)
+	slotMachine.PrepareRunner(ctx, mc)
+
+	slotMachine.AddInterfaceDependency(&catalog)
+	slotMachine.AddInterfaceDependency(&authService)
+
+	var vStateReportRecv = make(chan struct{})
+	slotMachine.PrepareMockedMessageSender(mc)
+	slotMachine.MessageSender.SendRole.SetCheckMessage(func(msg payload.Marshaler) {
+		res, ok := msg.(*payload.VStateReport)
+		require.True(t, ok)
+		require.Equal(t, payload.Missing, res.Status)
+		close(vStateReportRecv)
+	})
+
+	outgoing := slotMachine.GenerateLocal()
+
+	smExecute := SMExecute{
+		Payload: &payload.VCallRequest{
+			CallType:     payload.CTConstructor,
+			CallFlags:    payload.BuildCallFlags(contract.CallTolerable, contract.CallDirty),
+			CallOutgoing: outgoing,
+
+			Caller:         caller,
+			Callee:         class,
+			CallSiteMethod: "New",
+		},
+		Meta: &payload.Meta{
+			Sender: caller,
+		},
+	}
+	slotMachine.Start()
+	defer slotMachine.Stop()
+
+	smWrapper := slotMachine.AddStateMachine(ctx, &smExecute)
+	slotMachine.RunTil(smWrapper.BeforeStep(smExecute.stepStartRequestProcessing))
+	slotMachine.Migrate()
+	go slotMachine.RunTil(func(event debuglogger.UpdateEvent) bool {
```

Comment:
I don't like the idea for go routined runtil. I guess it can be really flacky. You can runtil Migration happened. And check that smexecute is finished without any messages.

Reply:
Add LeakTester and test with command
`go test -count=1000 -timeout 1200s -failfast -run=TestSendVStateReportWithMissingState_IfConstructorWasInterruptedBeforeRunnerCall github.com/insolar/assured-ledger/ledger-core/virtual/execute`